includes/admin/controllers/class-admin-listings.php

```diff
@@ -700,6 +702,32 @@ public function ajax_clear_payment_history() {
         wp_send_json_success( array( 'message' => _x( 'Listing\'s payment history successfully deleted', 'admin listings', 'business-directory-plugin' ) ) );
     }
 
+	/**
+	 * Assign a plan to a listing.
+	 * This assigns a plan to a listing in the admin backend.
+	 *
+	 * @since x.x
+	 */
+	public function ajax_assign_plan_to_listing() {
+		check_ajax_referer( 'wpbdp_ajax', 'nonce' );
+		$listing_id = wpbdp_get_var( array( 'param' => 'listing_id' ), 'post' );
+		$plan_id    = wpbdp_get_var( array( 'param' => 'plan_id' ), 'post' );
+		if ( ! $listing_id || ! $plan_id ) {
+			wp_send_json_error();
+		}
+		$listing = wpbdp_get_listing( $listing_id );
+		$current_plan = $listing->get_fee_plan();
+
+		if ( ! $current_plan || (int) $current_plan->fee_id != (int) $plan_id ) {
+			$payment = $listing->set_fee_plan_with_payment( $plan_id );
+
+			if ( $payment ) {
+				$payment->process_as_admin();
```

Comment:
Could this add a free payment to the listing or mark a pending listing as complete? Can you help me understand why the payment needs to be included here?

Reply:
Agreed, this would submit the payment which we do not want and can cause multiple payments saved with each plan change. I've removed the line after `$listing->set_fee_plan_with_payment( $plan_id );`

Reply:
It looks like that function creates a payment too. Is that needed?

Reply:
Changed it to use `set_fee_plan` which doesn't create a payment. Payment will only be created on submit but not on change

Reply:
Looks good!